File: v2/AssembleConsole.java
import java.util.Scanner;

public class AssembleConsole extends Assemble {
    private final Scanner sc = new Scanner(System.in);

    public void run() {
        execute();
    }

    @Override
    protected void print(String msg) {
        System.out.print(msg);
    }

    @Override
    protected void println(String msg) {
        System.out.println(msg);
    }

    @Override
    protected String readLine() {
        return sc.nextLine();
    }

    @Override
    protected void clear() {
        System.out.print(CLEAR_SCREEN);
        System.out.flush();
    }

    @Override
    protected void delay(int ms) {
        try {
            Thread.sleep(ms);
        } catch (InterruptedException ignored) {
        }
    }
}

File: python/assemble_console.py
import sys
import time

from assemble import CLEAR_SCREEN, Assemble


class AssembleConsole(Assemble):
    def run(self) -> None:
        self.execute()

    def print_(self, msg: str) -> None:
        print(msg, end="", flush=True)

    def println(self, msg: str) -> None:
        print(msg)

    def read_line(self) -> str | None:
        try:
            return input()
        except EOFError:
            return None

    def clear(self) -> None:
        sys.stdout.write(CLEAR_SCREEN)
        sys.stdout.flush()

    def delay(self, ms: int) -> None:
        time.sleep(ms / 1000.0)
